src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -80,43 +81,78 @@ public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposab
 
                 previousBroadcastTask = Task.WhenAll(tasks);
             });
+
+            channelName = _channelNamePrefix + ".AllExcept";
+            _logger.LogInformation("Subscribing to channel: {channel}", channelName);
+            _bus.Subscribe(channelName, async (c, data) =>
+            {
+                await previousBroadcastTask;
+
+                _logger.LogTrace("Received message from redis channel {channel}", channelName);
+
+                var message = DeserializeMessage<object[]>(data);
+                var invocationMessage = (InvocationMessage)message[0];
+                var excludedIds = (IReadOnlyList<string>)message[1];
+                var excludedIdsSet = new HashSet<string>(excludedIds);
+
+                //// TODO: This isn't going to work when we allow JsonSerializer customization or add Protobuf
+
+                var tasks = new List<Task>(_connections.Count);
+
+                foreach (var connection in _connections)
+                {
+                    if (!excludedIds.Contains(connection.ConnectionId))
+                    {
+                        tasks.Add(WriteAsync(connection, invocationMessage));
+                    }
+                }
+
+                previousBroadcastTask = Task.WhenAll(tasks);
+            });
         }
 
         public override Task InvokeAllAsync(string methodName, object[] args)
         {
             var message = new InvocationMessage(GetInvocationId(), nonBlocking: true, target: methodName, arguments: args);
 
-            return PublishAsync(typeof(THub).FullName, message);
+            return PublishAsync(_channelNamePrefix, message);
+        }
+
+        public override Task InvokeAllExceptAsync(string methodName, object[] args, IReadOnlyCollection<string> excludedIds)
+        {
+            var message = new InvocationMessage(GetInvocationId(), nonBlocking: true, target: methodName, arguments: args);
+            var data = new object[] { message, excludedIds };
```

Comment:
This looks... What was wrong with making a class that wraps InvocationMessage and adds a IReadOnlyCollection?